### PR TITLE
Debug pytorch gpu

### PR DESCRIPTION
**Pull request by giswqs**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by giswqs** [comment]
@h-vetinari Sorry for bothering you too much on this issue! You suggested adding `pytorch=*=cuda*` to the test requirements, how do we do it specifically? Should I follow the instructions at https://conda-forge.org/docs/maintainer/knowledge_base.html#cuda-builds? 

@Innixma The version bounds are temporarily removed so that you we debug which dependency is preventing the pytorch gpu version from being installed. Once we figure it out, we can put all the version bounds back in. Also, in terms of version release, conda-forge is bit more flexible than PyPI. We can change the dependency versions by bumping the build number without making a new release. Nothing needs to be changed in the autogluon source code. No new versions/builds are pushed to conda-forge until we resolve this issue.

**Comment by giswqs** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
> You suggested adding `pytorch=*=cuda*` to the test requirements, how do we do it specifically?

In
```
test:
  imports:
    - autogluon.multimodal
  commands:
    - python -c "from autogluon.multimodal import __version__ as v; assert v == '{{ version }}', f'{v} != {{ version }}'"
  requires:
    - pip
```
you have a `requires:` section, where you can specify testing-only requirements (usually pytest and other things that aren't resp. _shouldn't_ be a dependency of the regular package).

File: conda-forge.yml [hunk 1]
```diff
@@ -6,4 +6,10 @@ bot:
...
   osx_arm64: osx_64
-test: native_and_emulated
\ No newline at end of file
+test: native_and_emulated
+azure:
+  # flag for forcing the building all supported providers
+  force: true
+  # toggle for storing the conda build_artifacts directory (including the
+  # built packages) as an Azure pipeline artifact that can be downloaded
+  store_build_artifacts: true
```
Comment (h-vetinari) [inline conda-forge.yml:15]: You need to be very careful not to leave this in the PR once it gets merged to main.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,44 +22,44 @@ requirements:
...
     - python # >=3.7,<3.10
     - pip
   run:
     - python # >=3.7,<3.10
-    - numpy <1.24,>=1.21
-    - scipy <1.10.0,>=1.5.4
-    - pandas !=1.4.0,<1.6,>=1.2.5
-    - scikit-learn <1.2,>=1.0.0
-    - pillow <=9.4.0,>=9.3.0
+    - numpy >=1.21,<1.24
```
Comment (h-vetinari) [inline recipe/meta.yaml:31]: Can you remove the upper bound on numpy as well (even though I doubt that something depends on this already).

Comment (giswqs) [reply recipe/meta.yaml]: I have removed the numpy upper bound. Still getting the same error. 
> nptyping.error.InvalidArgumentsError: Unexpected argument '(typing.Any,)', expecting Shape[<ShapeExpression>] or Literal[<ShapeExpression>] or typing.Any.

File: conda-forge.yml (hunk 1 above)
Comment (giswqs) [reply conda-forge.yml]: I have removed the azure artifacts. I am now using docker on my computer to build and test the artifacts.

**Pull request closed without merging**